site/src/resources.rs

```diff
@@ -39,10 +41,23 @@ impl ResourceResolver {
         })
     }
 
-    pub fn get_static_asset(&self, path: &str) -> Option<Vec<u8>> {
-        StaticCompiledAssets::get(path)
-            .or_else(|| StaticAssets::get(path))
-            .map(|file| file.data.to_vec())
+    pub fn get_static_asset(&self, path: &str, allow_compression: bool) -> (Option<Vec<u8>>, bool) {
```

Comment:
Returning a tuple of the payload and a `bool` is ugly, and allows representing states that don't make sense (`(None, true)`). Could you please create an enum, e.g. `Payload`, which will have `Compressed` and `Uncompressed` variants, and then return `Option<Payload>`?